src/subsplit_dag.cpp

```diff
@@ -128,10 +132,10 @@ BitsetSizeMap SubsplitDAG::BuildGPCSPIndexer() const {
     SafeInsert(gpcsp_indexer, Bitset::PCSPOfPair(parent_subsplit, child_subsplit),
                gpcsp_idx);
   });
-  for (const auto &rootsplit : rootsplits_) {
-    const auto subsplit = rootsplit + ~rootsplit;
-    SafeInsert(gpcsp_indexer, subsplit, GetRootsplitIndex(subsplit));
-  }
+  IterateOverRootsplitIds([this, &gpcsp_indexer](size_t rootsplit_id) {
```

Comment:
Are you sure that we can't get rid of this sort of special purpose code by generalizing the traversals? I would have thought that we could add the rootsplit-associated edges to the traversal and get rid of this.